pex/pex.py

```diff
@@ -95,27 +95,38 @@ def interpreter(self):
         # type: () -> PythonInterpreter
         return self._interpreter
 
+    @property
```

Comment:
Thoughts on adding `@memoized_property` to Pex? This code will run twice iiuc.

Reply:
It caches manually in `self._envs` variable using the `if self._envs is None` guard.